File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Coleman-Liau index.
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // to determine the grade based on the index
    int grade = round(index);
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    int len = strlen(text);
    int letters = 0;
    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int counter = 0;
    bool track = false;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] != ' ' && track == false)
        {
            counter++;
            track = true;
        }
        else if (text[i] == ' ')
        {
            track = false;
        }
    }
    return counter;
}

int count_sentences(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)

    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            counter++;
        }
    }
    return counter;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = float(letters) / words * 100
    S = float(sentences) / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    grade = round(index)
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters


def count_words(text):
    counter = 0
    track = False
    for i in range(len(text)):
        if text[i] != ' ' and track == False:
            counter += 1
            track = True
        elif text[i] == ' ':
            track = False
    return counter


def count_sentences(text):
    counter = 0
    for i in range(len(text)):
        if text[i] == '!' or text[i] == '.' or text[i] == '?':
            counter += 1
    return counter


main()
